src/client/state/RoomsInput.js

```diff
@@ -200,6 +201,51 @@ class RoomsInput extends EventEmitter {
     return this.roomIdToInput.get(roomId)?.isSending || false;
   }
 
+  // Apply formatting to a plain text message
+  //
+  // This includes inserting any custom emoji that might be relevant, and (only if the
+  // user has enabled it in their settings) formatting the message using markdown.
+  formatAndEmojifyText(text) {
+    const allEmoji = getAllEmoji(this.matrixClient);
+
+    // Check to see if there are any :shortcode-style-tags: in the message
+    const shortcodes = Array.from(text.matchAll(/\B:([\w-]+):\B/g)).map((m) => m[1])
+      // Then filter to only the ones corresponding to a valid emoji
+      .filter((s) => allEmoji.has(s));
+    // But we'll put the emoji into the message after, so that they don't get escaped by
+    // the markdown formatter
+
+    // Now we apply markdown formatting
+    let formattedText;
+    if (settings.isMarkdown) {
+      formattedText = getFormattedBody(text);
+    } else {
+      formattedText = text;
+    }
+
+    // If there were any emojis, now we substitute :shortcodes: for the <img/> tag
+    for (let i = 0; i < shortcodes.length; i += 1) {
+      const shortcode = shortcodes[i];
+      const emoji = allEmoji.get(shortcode);
+      let tag;
+      if (emoji.mxc) {
+        tag = `<img data-mx-emoticon="" src="${
+          emoji.mxc
+        }" alt=":${
+          emoji.shortcode
+        }:" title=":${
+          emoji.shortcode
+        }:" height="32" />`;
+      } else {
+        tag = emoji.unicode;
+      }
+
+      formattedText = formattedText.replaceAll(`:${shortcode}:`, tag);
```

Comment:
This replaceAll will replace the alt="`:my_emoji:`" and title="`:my_emoji:`"  with the tag. if there are more then two emoji in raw text. 🤪
First emoji will be converted to tag and in next for loop turn it will also start replacing the value of above two property with the tag

Solution: 
the regex should only pick shortcode if: 
1. it has space around it. like:  `text = 'xyz :my_emoji: abc'`
2. or there is no text after/before it. like:  `text = 'xyz :my_emoji:'`, `text = ':my_emoji: abc'` or `text = ':my_emoji:'`

Reply:
How's this?  I think it is important that we be able to match emoji that are next to characters, such as if they're at the end of a quote or section of formatted text, like

> they said, "I love you :fancy-heart:"

or

> **\*\*wow. :shocked-pikachu:\*\***

but I was able to fix the bug by performing all replacements using the original match indices, which also fixed the inconsistencies between the two matchers.  The matcher still checks for word boundaries though, so something like this:isnt:right would neither match nor replace.